guides/compatibility-mode.md

```diff
@@ -0,0 +1,32 @@
+# Compatibility Mode
+Compatibility mode allows `@angular/material` to work with [Angular Material for AngularJS](https://material.angularjs.org/latest/).
+
+## Upgrade Process Overview
```

Comment:
I would omit this section, as I feel it should mostly be captured by upgrade guidance outside of material